# Random Location Finder Action->
#-> Provides interface to find random location of the next nearest 
#   unprocessed room from current robot location
#	when go to room location action server returns 
#	"False" result.
#-> input interface/msg format for action client
#	which is required by action server to process
#-> output interface/msg format to get the next
#	random goal pose.	 

#+++++++++++++++++++++++++++++++++++++goal definition++++++++++++++++++++++++++++++++++++++++++++++++++
sensor_msgs/Image input_img				# input map to check the accessible point
float64 map_resolution					# map resolution of the supplied map in meter
float64 map_origin_x					# x coordinate of origin of the supplied map in meter
float64 map_origin_y					# y coordinate of origin of the supplied map in meter
int32[] room_number						# total room number of the supplied map
int32[] room_min_x						# the minimum x-coordinate value of the room.
int32[] room_max_x						# the maximum x-coordinate value of the room.
int32[] room_min_y						# the minimum y-coordinate value of the room.
int32[] room_max_y						# the maximum y-coordinate value of the room.
int32 unsuccess_counter					# when unsuccess counter exceed the limit,random location finder action server set the room as a processed room
---
#+++++++++++++++++++++++++++++++++++++result definition+++++++++++++++++++++++++++++++++++++++++++++++++
sensor_msgs/Image output_img			# return the modified image->when unsuccess counter exceed the limit
int32 random_location_x					# provides x coordinate of the random location
int32 random_location_y					# provides y coordinate of the random location
---
#feedback definition